hefquin-engine/src/main/java/se/liu/ida/hefquin/engine/federation/access/impl/FederationAccessManagerBase1.java

```diff
@@ -161,28 +162,31 @@ protected Function<TPFResponse, CardinalityResponse> getFctToObtainCardinalityRe
 	protected static class FunctionToObtainCardinalityResponseFromSolMapsResponse implements Function<SolMapsResponse, CardinalityResponse>
 	{
 		public CardinalityResponse apply( final SolMapsResponse smResp ) {
-			final int cardinality = extractCardinality(smResp);
-			if ( cardinality < 0 )
+			final Integer cardinality;
+			try {
+				cardinality = extractCardinality( smResp );
+			}
+			catch ( final UnsupportedOperationDueToRetrievalError e ) {
```

Comment:
Related to [the other comment that I just wrote](https://github.com/LiUSemWeb/HeFQUIN/pull/410#discussion_r2046440844), here is the part where also the `IllegalArgumentException` from `extractCardinality( .. )` should be caught:
```suggestion
			catch ( final UnsupportedOperationDueToRetrievalError | IllegalArgumentException e ) {
```
... and then passed to the constructor of `CardinalityResponseImplWithoutCardinality`.